sdk/python/feast/feature_server.py

```diff
@@ -108,33 +114,26 @@ async def lifespan(app: FastAPI):
 
     app = FastAPI(lifespan=lifespan)
 
-    async def get_body(request: Request):
-        return await request.body()
-
     @app.post(
         "/get-online-features",
         dependencies=[Depends(inject_user_details)],
     )
-    async def get_online_features(body=Depends(get_body)):
-        body = json.loads(body)
-        full_feature_names = body.get("full_feature_names", False)
-        entity_rows = body["entities"]
+    async def get_online_features(request: GetOnlineFeaturesRequest) -> Dict[str, Any]:
```

Comment:
fastapi is smart about types. surfaces pydantic model in the autogenerated api docs